gc/base/standard/Scavenger.cpp

```diff
@@ -1543,6 +1546,33 @@ MM_Scavenger::copy(MM_EnvironmentStandard *env, MM_ForwardedHeader* forwardedHea
 			scavStats->_flipBytes += objectCopySizeInBytes;
 			scavStats->getFlipHistory(0)->_flipBytes[oldObjectAge + 1] += objectReserveSizeInBytes;
 		}
+		if (_extensions->scavengerScanOrdering == MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_DYNAMIC_BREADTH_FIRST) {
+			UDATA hotFieldOffset = _extensions->objectModel.getHotFieldOffset(forwardedHeader);
+			if (hotFieldOffset < UDATA_MAX) {
+				GC_SlotObject HotFieldObject(_omrVM, (fomrobject_t*)(destinationObjectPtr + hotFieldOffset));
+
+				omrobjectptr_t objectPtr = HotFieldObject.readReferenceFromSlot();							
+				if (NULL != objectPtr && isObjectInEvacuateMemory(objectPtr)) {
```

Comment:
```suggestion
				if ((NULL != objectPtr) && isObjectInEvacuateMemory(objectPtr)) {
```

Reply:
I belive you can remove NULL check altogether. if it is in NULL, it certainly won't be in Evacuate, so that the second check will fail.